File: Lucky_Ticket_Java/app/src/main/java/com/example/lucky_ticket_java/MainActivity.java
package com.example.lucky_ticket_java;

import androidx.appcompat.app.AppCompatActivity;
import android.graphics.Color;
import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    EditText number;
    ImageView lamp;
    Drawable drawable;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        number = findViewById(R.id.editText);
        lamp = findViewById(R.id.lamp);
        drawable = getDrawable(R.drawable.shape);
    }

    public void changeColor(View view){
        String num = number.getText().toString().trim();

        if (TextUtils.isEmpty(num) || num.length() != 6 ) {
            drawable.setTint(Color.GRAY);
            lamp.setBackground(drawable);
            Toast.makeText(getApplicationContext(), "incorrectly entered number", Toast.LENGTH_SHORT).show();
            return;
        }

        if(isLucky(num)){
            drawable.setTint(Color.GREEN);
            lamp.setBackground(drawable);
        }else {
            drawable.setTint(Color.RED);
            lamp.setBackground(drawable);
        }

    }

    static boolean isLucky(String num) {
        return sumOfDigits(num,0) == sumOfDigits(num,num.length() - 3);
    }

    static int sumOfDigits(String num, int start) {
        int sum = 0;
        for (int i = start; i < start + 3; ++i)
            sum += Character.digit(num.charAt(i), 10);
        return sum;
    }
}

File: Lucky_Ticket_Kotlin/app/src/main/java/com/example/lucky_ticket_kotlin/MainActivity.kt
package com.example.lucky_ticket_kotlin

import android.graphics.Color
import android.graphics.PorterDuff
import android.graphics.drawable.Drawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.widget.EditText
import android.widget.ImageView
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private var number: EditText? = null
    private var lamp: ImageView? = null
    private var drawable: Drawable? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        number = findViewById(R.id.editText)
        lamp = findViewById(R.id.lamp)
        drawable = getDrawable(R.drawable.shape)

    }

    fun changeColor(view: View){
        val num = number?.text.toString().trim { it <= ' ' }

        if (TextUtils.isEmpty(num) || num.length != 6) {
            drawable?.setTint(Color.GRAY)
            lamp?.background = drawable
            Toast.makeText(applicationContext, "incorrectly entered number", Toast.LENGTH_SHORT).show()
            return
        }

        if (isLucky(num)) {
            drawable?.setTint(Color.GREEN)
            lamp?.background = drawable
        } else {
            drawable?.setTint(Color.RED)
            lamp?.background = drawable
        }
    }

    fun isLucky(num: String): Boolean {
        return sumOfDigits(num, 0) == sumOfDigits(num, num.length - 3)
    }

    fun sumOfDigits(num: String, start: Int): Int {
        var sum = 0
        for (i in start until start + 3) sum += Character.digit(num[i], 10)
        return sum
    }
}